core/src/main/java/org/testcontainers/images/builder/ImageFromDockerfile.java

```diff
@@ -156,8 +168,24 @@ public ImageFromDockerfile withBuildArgs(final Map<String, String> args) {
         return this;
     }
 
+    /**
+     * Sets the Dockerfile to be used for this image. 
+     * It is recommended to use {@link #withDockerfile} instead because it honors .dockerignore
+     * files and therefore will be more efficient
+     * @param relativePathFromBuildRoot
+     */
     public ImageFromDockerfile withDockerfilePath(String relativePathFromBuildRoot) {
         this.dockerFilePath = Optional.of(relativePathFromBuildRoot);
         return this;
     }
+    
+    public ImageFromDockerfile withDockerfile(File dockerfile) {
+        this.dockerfile = Optional.of(dockerfile);
+        return this;
+    }
+    
+    public ImageFromDockerfile withDockerfile(String pathname) {
```

Comment:
let's not have this method, since it is unclear whether it will be using the classpath or FS

Reply:
Good point. It could also be a dockerfile as a string!

I don't see why we couldn't remove this method.

Reply:
I was thinking it may be convenient so people don't have to do `withDockerfile(new FIle("Dockerfile"))` in the normal case. Since it may cause confusion I'll remove it though